pkg/odo/cli/url/create.go

```diff
@@ -100,25 +100,27 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 		o.Context = genericclioptions.NewContext(cmd)
 	}
 
-	o.Client = genericclioptions.Client(cmd)
+	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
+		if !pushtarget.IsPushTargetDocker() {
+			o.Client = genericclioptions.Client(cmd)
 
-	routeSupported, err := o.Client.IsRouteSupported()
-	if err != nil {
-		return err
-	}
-	if routeSupported {
-		o.isRouteSupported = true
-	}
+			routeSupported, err := o.Client.IsRouteSupported()
+			if err != nil {
+				return err
+			}
+			if routeSupported {
```

Comment:
this is redundant, please directly assign `o.isRouteSupported, err :=  := o.Client.IsRouteSupported()`

Reply:
@girishramnani Fixed